### PR TITLE
Adding success_msg to assert

### PR DESCRIPTION
**Pull request by Im0**
##### SUMMARY
A minor change to allow a customized message on a successful assertion instead of `All assertions passed`.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
assert.py

##### ANSIBLE VERSION

```
ansible 2.7.0.dev0 (assert_success_message b2da29934c) last updated 2018/06/02 22:56:03 (GMT +1100)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/j/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/j/ansible/hacking/ansible/lib/ansible
  executable location = /home/j/ansible/hacking/ansible/bin/ansible
  python version = 2.7.12 (default, Nov 19 2016, 06:48:10) [GCC 5.4.0 20160609]
```

##### ADDITIONAL INFORMATION

Example playbook:

```
- hosts: localhost
  tasks:
    - name: Assert success message
      assert:
        that: "1 == 1"
        msg: "There was a failure {{ inventory_hostname }}"
        success_msg: "Successful message {{ inventory_hostname }}"
    - name: Assert success normal
      assert:
        that: "1 == 1"
        msg: "There was a failure {{ inventory_hostname }}"
    - name: Assert fail message
      assert:
        that: "1 != 1"
        msg: "There was a failure {{ inventory_hostname }}"
        success_msg: "Successful message {{ inventory_hostname }}"
```

Example output:

```
$ ansible-playbook test_assert.yml

PLAY [localhost] **********************************************************************************************************************************

TASK [Gathering Facts] ****************************************************************************************************************************
ok: [localhost]

TASK [Assert success message] *********************************************************************************************************************
ok: [localhost] => {
    "changed": false, 
    "msg": "Successful message localhost"
}

TASK [Assert success normal] **********************************************************************************************************************
ok: [localhost] => {
    "changed": false, 
    "msg": "All assertions passed"
}

TASK [Assert fail message] ************************************************************************************************************************
fatal: [localhost]: FAILED! => {
    "assertion": "1 != 1", 
    "changed": false, 
    "evaluated_to": false, 
    "msg": "There was a failure localhost"
}
	to retry, use: --limit @/home/johni/ansible/hacking/ansible/test_assert.retry

PLAY RECAP ****************************************************************************************************************************************
localhost                  : ok=3    changed=0    unreachable=0    failed=1   
```

### CODE REVIEW

File: lib/ansible/modules/utilities/logic/assert.py [hunk 1]
```diff
@@ -30,6 +30,9 @@
   msg:
     description:
       - "The customized message used for a failing assertion"
+  success_msg:
+    description:
+      - "The customized message used for a successful assertion"
```
Comment (bcoca) [inline lib/ansible/modules/utilities/logic/assert.py]: missing version added

File: lib/ansible/plugins/action/assert.py [hunk 2]
```diff
@@ -38,8 +38,11 @@ def run(self, tmp=None, task_vars=None):
             raise AnsibleError('conditional required in "that" string')
 
         msg = None
+        success_msg = None
         if 'msg' in self._task.args:
             msg = self._task.args['msg']
+        if 'success_msg' in self._task.args:
+            success_msg = self._task.args['success_msg']
```
Comment (bcoca) [inline lib/ansible/plugins/action/assert.py]: You should ensure it is a string (see string_types)

**Review by bcoca — CHANGES_REQUESTED** [review]
minor issues, other than that lgtm

File: lib/ansible/plugins/action/assert.py (hunk 2 above)
Comment (Im0) [reply lib/ansible/plugins/action/assert.py]: Thanks @bcoca , done.  Wondering if `that` input needs checking too.

File: lib/ansible/modules/utilities/logic/assert.py (hunk 1 above)
Comment (Im0) [reply lib/ansible/modules/utilities/logic/assert.py]: Thanks.  Done.

File: lib/ansible/plugins/action/assert.py (hunk 2 above)
Comment (bcoca) [reply lib/ansible/plugins/action/assert.py]: all parameters should, that is something we are revising now, but I'm not going to hold you up based on existing flaws, just trying to not add to the problem at this point.

**Review by bcoca — APPROVED** [review]

**Pull request merged**